lib/src/core/vehicle_plate/vehicle_plate.dart

```diff
@@ -0,0 +1,100 @@
+import 'package:persian_tools/src/constants/vehicle_plate/constants.dart';
+import 'package:persian_tools/src/core/vehicle_plate/model.dart';
+
+/// A wrapper class to have better access to plateDataset constant list
+abstract class _PlateDataset {
+  /// Returns car province name by the given key which is province code
+  static String? getCarProvince(int key) => plateDataset[0][key];
+
+  /// Returns motorcycle province name by the given key which is province code
+  static String? getMotorcycleProvince(int key) => plateDataset[1][key];
+
+  /// Returns category name by the given char
+  static String? getCategory(String key) => plateDataset[2][key];
+}
```

Comment:
Why should we have an abstract class here? We can turn its methods into the high-order private methods.